### ISSUE TITLE
notebooks/03: the ablation's ranking comes from the tied-embedding init, not from the blocks

### ISSUE THREAD

**Issue by mira687** [openlanguagemodel/openlanguagemodel#85]
`notebooks/03_custom_architecture_ablation_colab.ipynb` runs end to end and prints two numbers,
and those two numbers are the whole notebook:

```
MHA + LayerNorm + ClassicFFN: params=6,829,952, one-step loss=125.7128
GQA + RMSNorm + SwiGLU: params=6,925,312, one-step loss=128.9006
```

A reader concludes MHA + LayerNorm + ClassicFFN is the better block by 3.2 nats. I do not think
that conclusion survives the init.

Both losses are ~12x chance (`ln(50257) = 10.8249`) because `TinyAblationLM` ends in
`OutputHead(embed_dim, vocab_size, tied_embedding=embedding)` and `Embedding` still carries
`nn.Embedding`'s default `normal_(0, 1)` — #65, closed, still on `main`. Embedding weight std
**1.0000**, logits at init std **11.33**, absmax **148.2**.

Re-initing that one tied matrix to `normal_(0, 0.02)` (the workaround in #65) and changing nothing
else, over five seeds:

| seed | shipped MHA | shipped GQA | winner | re-init MHA | re-init GQA | winner |
|---|---|---|---|---|---|---|
| 7 (notebook default) | 125.7128 | 128.9006 | MHA | 10.9017 | 10.8037 | GQA |
| 1 | 120.8146 | 125.3466 | MHA | 10.7448 | 10.8941 | MHA |
| 2 | 124.2578 | 128.8893 | MHA | 10.7613 | 10.8237 | MHA |
| 3 | 125.2859 | 128.4196 | MHA | 10.7969 | 10.8421 | MHA |
| 4 | 117.6968 | 119.7965 | MHA | 10.9840 | 10.8382 | GQA |

As shipped the gap is 2.1–4.6 nats and MHA wins every seed, which reads as a real architectural
effect. With a sane init the gap is 0.04–0.15 nats around chance and the winner flips with the
seed. So what the notebook measures consistently is which block's init amplifies a unit-scale
embedding slightly less.

Second thing, independent of the init: the logged step-1 loss is the forward loss from **before**
the optimizer step. An independent `no_grad` cross-entropy on the freshly built model gives
125.7128 / 128.9006 — the same four decimals as the trainer's "one-step loss". So
`max_steps=1` measures initialization by construction, and a fixed init would leave the notebook
comparing two models that have not trained. Both re-init rows above sit at chance, as expected.

The Perplexity column prints `485165195.41` for both rows — `exp(min(loss, 20))` saturated, same
as #65.

This is the part my earlier comment on #65 left open. I said there that the unit-scale init might
be "purely a first-run-experience problem rather than a training one". In this notebook it is not
cosmetic: it changes a printed conclusion in a teaching artifact.

**How it was run.** Notebook cells executed in order in one process, pip cell skipped. Python
3.12.14, `openlanguagemodel==2.2.1` from PyPI (`src/olm/nn/embeddings/token_embed.py` byte-identical
to `main`), `transformers==4.57.1`, `numpy==1.26.4`, torch 2.2.2, CPU. On torch < 2.3 I had to
backfill `torch.amp.GradScaler` from `torch.cuda.amp` before any `olm.train` import — that is #76,
not this. Colab's default torch does not need it.

**Not checked:** whether a longer run separates the two blocks at all, whether either block is
actually better, and whether 02/04 change conclusions the same way. 04 shows the same init scale
(loss 118.8 → 112.4 over 3 steps, perplexity pinned at the clamp) but its output is a strategy
summary rather than a comparison, so nothing it prints is wrong. 02 runs end to end on CPU in 47s
and correctly skips the 125M training.
